tests/Behat/WysiwygContext.php

```diff
@@ -0,0 +1,354 @@
+   *   The label of the field containing the WYSIWYG editor.
+   *
+   * @When I enter :text in the :label wysiwyg editor
+   */
+  public function enterTextInWysiwyg($text, $label) {
+    // If we are running in a JavaScript enabled browser, first click the
+    // 'Source' button so we can enter the text as HTML and get the same result
+    // as in a non-JS browser.
+    if ($this->browserSupportsJavaScript()) {
+      $this->pressWysiwygButton($label, 'Source');
+      $this->setWysiwygText($label, $text);
+    }
+    else {
+      $this->getSession()->getPage()->fillField($label, $text);
+    }
+  }
+
+  /**
+   * Insert link to node through wysiwyg and DrupalLink button.
+   *
+   * @param string $node_title
+   *   Title of the node.
+   * @param string $field
+   *   Field title of the node.
+   *
+   * @When I insert link to :node_title through wysiwyg editor in :field field
+   */
+  public function iInsertDrupalLink(string $node_title, string $field) {
+
+    $session = $this->getSession();
+    $page = $session->getPage();
+
+    $this->pressWysiwygButton($field, 'Link (Ctrl+K)');
+
+    $node1 = $this->getNodeByTitle($node_title);
+
+    $this->waitForAjaxToFinish();
+
+    $href_field = $page->findField('attributes[href]');
+    $href_field->hasAttribute('data-autocomplete-path');
+    // Trigger a keydown event to active a autocomplete search.
+    $href_field->setValue($node1->label());
+    $href_field->keyDown(' ');
+
+    $this->getSession()->wait(5000, "jQuery('.linkit-result-line.ui-menu-item').length > 0");
+
+    // Find the first result and click it.
+    $page->find('xpath', '//li[contains(@class, "linkit-result-line") and contains(@class, "ui-menu-item")][1]')->click();
+
+    $page->find('css', '.editor-link-dialog button:contains("Save")')->click();
+
+    $this->waitForAjaxToFinish();
+  }
+
+  /**
+   * Check link to node.
+   *
+   * @param string $node_title
+   *   Title of the node.
+   * @param string|null $langcode
+   *   Langcode if applicable.
+   *
+   * @throws \Drupal\Core\Entity\EntityMalformedException
+   *
+   * @Then I should see link to :node_title node
+   */
+  public function assertProcessedPurlLink(string $node_title, $langcode = FALSE) {
+    $session = $this->getSession();
+    $page = $session->getPage();
+
+    $node1 = $this->getNodeByTitle($node_title);
+
+    if (!empty($langcode)) {
+      $node1 = $node1->getTranslation($langcode);
+    }
+
+    $lang_prefix = $langcode ? '/' . $langcode : '';
+
+    $node1_url = '/' . $this->getDrupalParameter('drupal')['drupal_root'] . $lang_prefix . $node1->toUrl()->toString();
+
+    $count = count($page->findAll('css', 'a[href="' . $node1_url . '"]'));
+
+    switch (1) {
+      case $count === 0:
+        throw new \Exception("Could not find link '$node1_url' to relevant node.");
+
+      case $count > 1:
+        throw new \Exception("Multiple link '$node1_url' to relevant node found.");
+    }
+  }
+
+  /**
+   * Set alias of the node.
```

Comment:
There are methods such as this which do not belong in the WYSIWYG context right? Let's move them to the Feature or Drupal context.